src/adaptors/amqp/qd_connector.c

```diff
@@ -473,21 +451,27 @@ void qd_connector_remote_opened(qd_connector_t *connector)
 /**
  * Set the child connection of the connector
  */
-void qd_connector_add_connection(qd_connector_t *connector, qd_connection_t *ctx)
+void qd_connector_add_connection(qd_connector_t *connector, qd_connection_t *qd_conn)
 {
-    assert(ctx->connector == 0);
-
+    assert(qd_conn->connector == 0);
     sys_atomic_inc(&connector->ref_count);
-    ctx->connector = connector;
-    connector->qd_conn = ctx;
+    qd_conn->connector = connector;
+    connector->qd_conn = qd_conn;
+    if (!connector->is_data_connector) {
```

Comment:
What does this mean that it is not a data connector? Could we have a comment to explain?

Reply:
Oh, I see. It means that it's a control connector.

Never mind.